### ISSUE TITLE
Crash ``'Attribute' object has no attribute 'name'`` when accessing private member of private member

### ISSUE THREAD

**Issue by rcfox** [PyCQA/pylint#6749]
### Bug description

When parsing the following file:

```python
class Foo:
    __bar = None

    def test(self):
        if Foo.__bar.__qux:
            pass

    def __init__(self):
        self.__qux = 1
```

pylint crashed with a ``AttributeError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1121, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1156, in _check_file
    check_astroid_module(ast_node)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1309, in check_astroid_module
    ast_node, walker, rawcheckers, tokencheckers
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1355, in _check_astroid_module
    walker.walk(node)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 76, in walk
    self.walk(child)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 78, in walk
    callback(astroid)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/checkers/classes/class_checker.py", line 877, in leave_classdef
    self._check_unused_private_attributes(node)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/checkers/classes/class_checker.py", line 996, in _check_unused_private_attributes
    and attribute.expr.name == "self"
AttributeError: 'Attribute' object has no attribute 'name'
```

**Note**: It only seems to happen if `self.__qux` is defined on `Foo`.

### Command used

```shell
pylint foo.py
```

### Pylint output

```shell
************* Module foo
foo.py:1:0: C0104: Disallowed name "foo" (disallowed-name)
foo.py:1:0: C0114: Missing module docstring (missing-module-docstring)
foo.py:1:0: C0115: Missing class docstring (missing-class-docstring)
foo.py:2:4: C0116: Missing function or method docstring (missing-function-docstring)
foo.py:3:11: W0212: Access to a protected member __qux of a client class (protected-access)
foo.py:3:11: E1101: Class 'Foo' has no '__bar' member (no-member)
foo.py:2:4: R0201: Method could be a function (no-self-use)
Exception on node <ClassDef.Foo l.1 at 0x7f3ca8847e50> in file '/home/rfox/foo.py'
Traceback (most recent call last):
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 78, in walk
    callback(astroid)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/checkers/classes/class_checker.py", line 877, in leave_classdef
    self._check_unused_private_attributes(node)
  File "/home/rfox/.venvs/kepler/lib/python3.7/site-packages/pylint/checkers/classes/class_checker.py", line 996, in _check_unused_private_attributes
    and attribute.expr.name == "self"
AttributeError: 'Attribute' object has no attribute 'name'
foo.py:1:0: F0001: Fatal error while checking '/home/rfox/foo.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/home/rfox/.cache/pylint/pylint-crash-2022-05-30-06.txt'. (fatal)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

I won't claim that this is good code, but pylint should provide a useful error instead of crashing.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.7.9+ (heads/3.7:db95802, Nov 24 2020, 15:33:29)
```

### OS / Environment

Ubuntu Linux 20.04.4

**Comment by jacobtylerwalls** [comment]
Duplicate of #6709, to be released shortly in 2.14!

**Comment by rcfox** [comment]
Ah, sorry. I did search for duplicates, but didn't think to search closed tickets. Thanks!
